src/dali_executor/dali_executor.cc

```diff
@@ -136,6 +134,32 @@ bool DaliExecutor::IsNoCopy(device_type_t es_device, const IDescr& input) {
           input.buffers[0].device_id == pipeline_.DeviceId());
 }
 
+
+static bool streq(const char * lhs, const char * rhs) {
+  return strcmp(lhs, rhs) == 0;
+}
+
+
+bool DaliExecutor::IsInputConsumed() {
+  for (auto &name : input_names_) {
+    auto trace = pipeline_.TryGetOperatorTrace(name, "depleted");
+    if (!trace.has_value()) {
+      throw std::runtime_error("DALI Error: 'depleted' trace shall be available for every input operator.");
```

Comment:
You're not stating requirements, but rather producing diagnostics. Mentioning the specific name of the operator would be nice, too.
```suggestion
      throw std::logic_error("DALI Error: \"depleted\" trace not found for input \"" + name +"\". "
                             "It must be available for every input operator.");
```

Reply:
Done